streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -813,6 +727,10 @@ private void updateProcessorContext(final StampedRecord record, final ProcessorN
      * Currently only changelog topic offsets need to be checkpointed.
      */
     private Map<TopicPartition, Long> checkpointableOffsets() {
+        if (state() == State.RESTORING) {
```

Comment:
Should we return emptyMap if we are SUSPENDED as well?

Reply:
EDIT: actually, I think we need to accumulate the consumed offsets when we just transited to suspend and then called prepareCommit, but if we are already in suspended then it is actually okay to return an emptyMap. However since we do not know if we have just transited to suspended and the below code should not be a big overhead, we can just keep it as-is.

So please ignore my previous comment :)